### ISSUE TITLE
1E39 bug carried over from C64

### ISSUE THREAD

**Issue by tomxp411** [X16Community/x16-rom#247]
Browsing YouTube today, I found a video about the VAL function and self destructing programs. This bug still exists in Commander X16 BASIC

https://www.youtube.com/watch?v=6j8kel86F60

The statement:
`10 PRINT VAL("1E39") : REM MORE STUFF` will cause anything after the 9 to be _removed_ from the program line when executed.

```
10 PRINT VAL("1E39"):REM TEST
20 REM LINE 20
READY.
RUN
?OVERFLOW ERROR IN 10
READY.
LIST
10 PRINT VAL("1E39
20 REM LINE 20
READY.
```
The same thing happens if you assign a$ as a variable: 

```
10 A$="1E39":REM TEST
20 PRINT VAL(A$):REM MORE TEXT
becomes
10 A$="1E39
20 PRINT VAL(A$):REM MORE TEXT
```
This has apparently been fixed in the Commodore 128 and in the 264 series. So maybe porting the function from the Plus 4 or 128 can correct this issue.

**Comment by calmsacibis995** [comment]
It does need to be adapted for the X16, as during my experimentation with replacing the existing routines with the ones from the TED mahcines, I got this message on the emulator:

`Warning: encountered Rockwell instruction BBS at $04:df02.
	Future Commander X16 hardware may ship with a 65C816 CPU,
	which does not support these instructions.
	This will be the only warning given for Rockwell
	instructions until the emulator is relaunched.
	Pass -rockwell to the command line to suppress this warning.`

**Comment by calmsacibis995** [comment]
OK, some progress has been made, but it's still not perfect, as you can see, the last character of the comment is cut off.

![2024-08-31_23-05](https://github.com/user-attachments/assets/eb6882c4-75a9-48ef-a394-cdffe58e5be3)

**Comment by mooinglemur** [comment]
Closed by #357
